lms/djangoapps/discussion/tasks.py

```diff
@@ -32,6 +33,30 @@
 ROUTING_KEY = getattr(settings, 'ACE_ROUTING_KEY', None)
 
 
+@task(base=LoggedTask)
+def update_discussions_map(context):
+    course_key = CourseKey.from_string(context['course_id'])
+    discussions_id_map = {}
+
+    with modulestore().bulk_operations(course_key):
+        course = modulestore().get_course(course_key, depth=None)
+        blocks_stack = [course]
+        while blocks_stack:
+            curr_block = blocks_stack.pop()
+            children = curr_block.get_children() if curr_block.has_children else []
+            if (
+                curr_block.category == 'discussion' and
+                hasattr(curr_block, 'discussion_id') and
+                curr_block.discussion_id
+            ):
+                discussions_id_map[curr_block.discussion_id] = unicode(curr_block.scope_ids.usage_id)
+
+            # Add this blocks children to the stack so that we can traverse them as well.
+            blocks_stack.extend(children)
+
+    # TODO Store discussions_id_map for later retrieval
```

Comment:
@ormsbee I know you and @nasthagiri discussed storing this in S3. I was interested in understanding the reason for S3 as opposed to a mysql database table with columns for `discussion_id` and `course_id`?

Reply:
S3 was my first take because these are mostly written and queried in batch, but I feel like we've already spent more time on this than it really deserves, so I'm not going to object to a Django model if it makes things easier.